sphinx/builders/latex/__init__.py

```diff
@@ -220,6 +225,37 @@ def write(self, *ignored):
             docwriter.write(doctree, destination)
             logger.info("done")
 
+    def get_doc_context(self, title, author, docclass):
+        # type: (unicode, unicode, unicode) -> Dict[unicode, Any]
+        context = DEFAULT_SETTINGS.copy()
+
+        # Add special settings for latex_engine
+        context.update(ADDITIONAL_SETTINGS.get(self.config.latex_engine, {}))
+
+        # for xelatex+French, don't use polyglossia by default
+        if self.config.latex_engine == 'xelatex':
+            if self.config.language:
+                if self.config.language[:2] == 'fr':
+                    context['polyglossia'] = ''
+                    context['babel'] = r'\usepackage{babel}'
+
+        # Apply user settings to context
+        context.update(self.config.latex_elements)
+        context['release'] = self.config.release
+
+        # Override settings by latex_documents
+        context['title'] = title
+        context['author'] = author
+        context['use_xindy'] = self.config.latex_use_xindy
+
+        # for compatibilities
```

Comment:
Now we can translate message on templates directly. So I move them to templates. It means users can't modify the message by `latex_elements`. I think it is not expected feature.

Note: I have to add them to CHANGES entry.